### ISSUE TITLE
Add data augmentation to PyTorch backends

### ISSUE THREAD

**Issue by lewfish** [azavea/raster-vision#831]
We would like to add an option for training data augmentation to the new PyTorch backends in PR #830 . A good start would be adding it to the chip classification backend. This is roughly where the code should go: https://github.com/azavea/raster-vision/blob/c1594b211ed3cd267b451248e08a50a70f35cc1c/rastervision/backend/torch_utils/chip_classification/data.py#L16-L20

Note this link might not work once this PR is merged into master. 

We should use the albumentations library https://github.com/albu/albumentations. Here is an example of using it for object detection from another codebase (which formed the basis of the object detection backend in RV): https://github.com/lewfish/mlx/blob/88e21d9fdd2f0d7b3effb750a65b64e308bccc08/mlx/od/data.py#L154-L173

**Comment by lewfish** [comment]
@lmbak

**Comment by lmbak** [comment]
Work in progress at https://github.com/lmbak/raster-vision/tree/lmb/augmentation

**Comment by lmbak** [comment]
@lewfish I am currently testing, but run into the following error:

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/src/rastervision/__main__.py", line 17, in <module>
    rv.main()
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/src/rastervision/cli/main.py", line 294, in run_command
    rv.runner.CommandRunner.run(command_config_uri)
  File "/opt/src/rastervision/runner/command_runner.py", line 11, in run
    CommandRunner.run_from_proto(msg)
  File "/opt/src/rastervision/runner/command_runner.py", line 17, in run_from_proto
    command.run()
  File "/opt/src/rastervision/command/train_command.py", line 21, in run
    task.train(tmp_dir)
  File "/opt/src/rastervision/task/task.py", line 137, in train
    self.backend.train(tmp_dir)
  File "/opt/src/rastervision/backend/pytorch_chip_classification.py", line 225, in train
    train_uri)
  File "/opt/src/rastervision/backend/pytorch_chip_classification.py", line 74, in make_debug_chips
    _make_debug_chips('train')
  File "/opt/src/rastervision/backend/pytorch_chip_classification.py", line 61, in _make_debug_chips
    for i, (x, y) in enumerate(ds):
  File "/opt/conda/lib/python3.6/site-packages/torchvision/datasets/folder.py", line 140, in __getitem__
    sample = self.transform(sample)
  File "/opt/conda/lib/python3.6/site-packages/torchvision/transforms/transforms.py", line 61, in __call__
    img = t(img)
  File "/opt/conda/lib/python3.6/site-packages/albumentations/core/transforms_interface.py", line 55, in __call__
    params = self.update_params(params, **kwargs)
  File "/opt/conda/lib/python3.6/site-packages/albumentations/core/transforms_interface.py", line 103, in update_params
    {'cols': kwargs['image'].shape[1],
KeyError: 'image'

```

It seems the albumentations library expects a keyword-argument 'image' for calling one of the transformers. But the transformers seem to be called from the torchvision library, which does not supply this keyword argument. But this does not seem to be the case in the code you mentioned https://github.com/lewfish/mlx/blob/88e21d9fdd2f0d7b3effb750a65b64e308bccc08/mlx/od/data.py#L154-L173

It seems you also use the same version of albumentations in your code (0.3.* vs 0.3.3). For RV torchvision.__version__=0.4.0, I could not find the version used in your above code example.

Is it correct to conclude that this is something that cannot be handled directly from within RV? I.e. We cannot determine how torchvision calls its transformers, right? Or am I missing something?

**Comment by lmbak** [comment]
Found the problem. The albumentations library expects a numpy array, while the databunch contains PIL images.

**Comment by lewfish** [comment]
Done for all backends in RV2.
